src/wifi-menu

```diff
@@ -242,13 +242,27 @@ cd /  # We do not want to spawn anything that can block unmounting
 
 INTERFACE=$1
 if [[ -z "$INTERFACE" ]]; then
-    INTERFACE=(/sys/class/net/*/wireless/)
-    if [[ ${#INTERFACE[@]} -ne 1 || ! -d "$INTERFACE" ]]; then
+    INTERFACES=(/sys/class/net/*/wireless/)
+    if [[ ${#INTERFACES[@]} -eq 1 && -d "$INTERFACES" ]]; then
+        INTERFACE=${INTERFACES:15:-10}
+    elif [[ ${#INTERFACES[@]} -gt 1 ]]; then
+        report_debug "Multiple wireless interfaces found: '${INTERFACES[@]}'"
+	WIRELESS_INTERFACES=()
+	for WIRELESS_IDX in ${!INTERFACES[@]}; do
+	  if [[ -d ${INTERFACES[$WIRELESS_IDX]} ]]; then
+	      echo $WIRELESS_IDX ${INTERFACES[$WIRELESS_IDX]}
+	      WIRELESS_INTERFACES+=($WIRELESS_IDX ${INTERFACES[$WIRELESS_IDX]:15:-10})
+	  fi
+	done
+	echo Interface: ${WIRELESS_INTERFACES[@]}
+        IFACE_CHOICE=$(dialog --menu "Select wireless interface you wish to use:" 24 50 12 "${WIRELESS_INTERFACES[@]}" --stdout)
+        INTERFACE=${INTERFACES[$IFACE_CHOICE]:15:-10}
+        report_debug "Wireless interface selected: '$INTERFACE'"
+    else
         report_error "Invalid interface specification"
         usage
         exit 255
     fi
-    INTERFACE=${INTERFACE:15:-10}
```

Comment:
While I like the idea, let's go with a slightly different logic. Here is the gist of what I think is a more elegant implementation:
```
if [[ ${#INTERFACES[@]} -gt 1 ]]; then
    INTERFACE=$(dialog --no-items --menu "Text:" 24 50 12 $(for if in ${INTERFACES[@]}; do echo ${if:15:-10}; done) --stdout)
elif [[ -d "$INTERFACES" ]]; then
    INTERFACE=${INTERFACES:15:-10}
else
    report_error etc.
```
Sure, that second line is a bit too much this way, but the point is that interface names are unique, so we can simply use `--no-items` here. Additionally, the `-d` check is only because we are too lazy to set `nullglob` (plus, it catches the case where there are no interfaces at all).

Reply:
Excellent. I have only limited bash experience and no dialog experience, so it is nice to what the proper way to do things is.

Reply:
Ok. This has been implemented. Tested on 0, 1, and 2 interfaces.